lldb/tools/debugserver/source/debugserver.cpp

```diff
@@ -208,27 +228,8 @@ RNBRunLoopMode RNBRunLoopLaunchInferior(RNBRemote *remote,
   // figure our how we are going to launch automatically.
 
   nub_launch_flavor_t launch_flavor = g_launch_flavor;
-  if (launch_flavor == eLaunchFlavorDefault) {
-    // Our default launch method is posix spawn
-    launch_flavor = eLaunchFlavorPosixSpawn;
-
-#if defined WITH_FBS
-    // Check if we have an app bundle, if so launch using BackBoard Services.
-    if (is_dot_app(inferior_argv[0])) {
-      launch_flavor = eLaunchFlavorFBS;
-    }
-#elif defined WITH_BKS
-    // Check if we have an app bundle, if so launch using BackBoard Services.
-    if (is_dot_app(inferior_argv[0])) {
-      launch_flavor = eLaunchFlavorBKS;
-    }
-#elif defined WITH_SPRINGBOARD
-    // Check if we have an app bundle, if so launch using SpringBoard.
-    if (is_dot_app(inferior_argv[0])) {
-      launch_flavor = eLaunchFlavorSpringBoard;
-    }
-#endif
-  }
+  if (launch_flavor == eLaunchFlavorDefault)
+    launch_flavor = default_launch_flavor(inferior_argv[0]);
```

Comment:
I extracted this function to put all the launch flavor logic together, and made `is_dot_app` a lambda within it.

Reply:
Thanks! Is this swift-specific code? I feel like even if this isn't on llvm.org it probably should got there...

Reply:
It doesn't appear to be swift specific, it seems more Apple specific. I'll bring it up with @jasonmolenda.

Reply:
Yeah, it should be landed on llvm.org monorepo.

Reply:
is it ok to commit this fix, and follow up with a move to llvm?

Reply:
seems like it's ok to commit and follow up (based on https://github.com/apple/llvm-project/pull/1749#discussion_r485131646)

Reply:
moved to https://reviews.llvm.org/D87327